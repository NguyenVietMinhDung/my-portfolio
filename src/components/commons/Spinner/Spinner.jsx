import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import styles from './Spinner.scss';

const Spinner = () => (
  <div styleName="spinner">
    <div styleName="background">
      <div styleName="animated-part" />
    </div>
  </div>
);

export default memo(CSSModules(Spinner, styles));
