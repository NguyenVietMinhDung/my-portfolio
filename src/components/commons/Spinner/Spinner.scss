@import '../../../styles/abstracts/variables';
@import '../../../styles/abstracts/mixins';

.spinner {
	.background {
		display: flex;

		width: 100%;
		height: 100vh;
		align-items: center;
		flex-flow: row wrap;
		justify-content: center;
	}

	.animated-part {
		width: 60px;
		height: 60px;

		animation: spin 2s linear infinite;

		border: 8px solid $white-smoke-color;
		border-top: 8px solid $violet-color;
		border-radius: 50%;
	}

	@keyframes spin {
		0% {
			transform: rotate(0deg);
		}

		100% {
			transform: rotate(360deg);
		}
	}
}
