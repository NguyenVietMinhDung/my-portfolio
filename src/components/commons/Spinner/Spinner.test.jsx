import React from 'react';
import { shallow } from 'enzyme';
import Spinner from './Spinner';

describe('<Spinner />', () => {
  test('renders correctly', () => {
    const component = shallow(<Spinner />);
    expect(component).toMatchSnapshot();
  });
});
