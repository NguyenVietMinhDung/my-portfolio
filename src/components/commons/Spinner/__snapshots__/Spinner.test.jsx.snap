// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders correctly 1`] = `
<div
  className="spinner"
>
  <div
    className="background"
  >
    <div
      className="animated-part"
    />
  </div>
</div>
`;
